app/src/main/java/protect/card_locker/LoyaltyCard.java

```diff
@@ -10,30 +11,88 @@
 import java.util.Currency;
 import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
 
 public class LoyaltyCard implements Parcelable {
-    public final int id;
-    public final String store;
-    public final String note;
+    public int id;
+    public String store;
+    public String note;
     @Nullable
-    public final Date validFrom;
+    public Date validFrom;
     @Nullable
-    public final Date expiry;
-    public final BigDecimal balance;
+    public Date expiry;
+    public BigDecimal balance;
     @Nullable
-    public final Currency balanceType;
-    public final String cardId;
+    public Currency balanceType;
+    public String cardId;
     @Nullable
-    public final String barcodeId;
+    public String barcodeId;
     @Nullable
-    public final CatimaBarcode barcodeType;
+    public CatimaBarcode barcodeType;
     @Nullable
-    public final Integer headerColor;
-    public final int starStatus;
-    public final int archiveStatus;
-    public final long lastUsed;
+    public Integer headerColor;
+    public int starStatus;
+    public int archiveStatus;
+    public long lastUsed;
     public int zoomLevel;
 
+    public static final String BUNDLE_LOYALTY_CARD_ID = "loyaltyCardId";
+    public static final String BUNDLE_LOYALTY_CARD_STORE = "loyaltyCardStore";
+    public static final String BUNDLE_LOYALTY_CARD_NOTE = "loyaltyCardNote";
+    public static final String BUNDLE_LOYALTY_CARD_VALID_FROM = "loyaltyCardValidFrom";
+    public static final String BUNDLE_LOYALTY_CARD_EXPIRY = "loyaltyCardExpiry";
+    public static final String BUNDLE_LOYALTY_CARD_BALANCE = "loyaltyCardBalance";
+    public static final String BUNDLE_LOYALTY_CARD_BALANCE_TYPE = "loyaltyCardBalanceType";
+    public static final String BUNDLE_LOYALTY_CARD_CARD_ID = "loyaltyCardCardId";
+    public static final String BUNDLE_LOYALTY_CARD_BARCODE_ID = "loyaltyCardBarcodeId";
+    public static final String BUNDLE_LOYALTY_CARD_BARCODE_TYPE = "loyaltyCardBarcodeType";
+    public static final String BUNDLE_LOYALTY_CARD_HEADER_COLOR = "loyaltyCardHeaderColor";
+    public static final String BUNDLE_LOYALTY_CARD_STAR_STATUS = "loyaltyCardStarStatus";
+    public static final String BUNDLE_LOYALTY_CARD_ARCHIVE_STATUS = "loyaltyCardArchiveStatus";
+    public static final String BUNDLE_LOYALTY_CARD_LAST_USED = "loyaltyCardLastUsed";
+    public static final String BUNDLE_LOYALTY_CARD_ZOOM_LEVEL = "loyaltyCardZoomLevel";
+
+    /**
+     * Create a loyalty card object with default values
+     */
+    public LoyaltyCard() {
+        this.id = -1;
+        this.store = "";
+        this.note = "";
+        this.validFrom = null;
+        this.expiry = null;
+        this.balance = new BigDecimal("0");
+        this.balanceType = null;
+        this.cardId = "";
+        this.barcodeId = null;
+        this.barcodeType = null;
+        this.headerColor = null;
+        this.starStatus = 0;
+        this.lastUsed = 0;
```

Comment:
I think 0 is better as it's deterministic. But the previous default for new cards used the current time. Not sure that matters?

Reply:
I think it makes sense to use the current time as "last used" time when creating a new card by default, that way it doesn't have to be set everywhere all the time